back/controllers/rangeSupervision.js

```diff
@@ -34,13 +39,57 @@ const controller = {
   },
 
   create: async function createSupervision(request, response) {
+    
+    try{
+      //fetches the supervisor id who is assgined to the range.
+      const vastaanottaja = await geetUserEmail(response.req.body.supervisor);
+      //knex part that should be done in models? returns the email based on the id fetched above:
+      async function geetUserEmail(key) {
+        return await knex
+          .from('user')
+          .where({ 'user.id': key })
+          .select('user.email');
+      }
```

Comment:
I would never recommend having function definitions inside functions. Either move the definition to the start of the file or into the models. If you move it to models don't worry whether it's the "right file" for it: just put it into rangeSupervision.js because this file uses it.

Reply:
Also: geet -> get ?

Reply:
Done